### PR TITLE
 Fixed unnecessarily long freezes when switching simulcast streams.

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** https://github.com/aws/amazon-chime-sdk-js/issues/2374 

**Description of changes:**

* Fix `active` check in demo which unnecessarily hides videos during rebind. Upon inspection...I found `active` to be quite arbitrarily calculated so I added some recommendations to docstring.
* Use `subscribeFrameSent` on remote video updates so we will cache the previous index and be able to skip subscribes on simulcast stream switches when senders have switched layers.

Note i have made some backend changes too, which should be out within a week or so, but those are not coupled to these changes, just solve similar problems.

**Testing:**
Ignoring backend fixes, these two changes can be verified with the following reduced steps.

* Start 2 attendees, one with simulcast. No other settings need to be enabled.
* Start sending on the simulcast attendee.
* Join the conference with a 3rd attendee.
* Leave the conference with the 3rd attendee.
* Verify no pauses (e.g. 1 second or longer, ofc depends on RTT).

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*

See steps above.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
n

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ltrung — APPROVED** [review]

File: src/videotile/VideoTileState.ts [hunk 1]
```diff
@@ -26,7 +26,9 @@ export default class VideoTileState {
   isContent: boolean = false;
 
   /**
-   * Indication of whether the tile has active video stream.
+   * Indication of whether the tile has active video stream. This value may be somewhat arbitrary in how it is calculated,
+   * it is recommended to used the more specific values (e.g. `paused`, `boundVideoElement`, and `boundVideoStream`) to
```
Comment (devalevenkatesh) [inline src/videotile/VideoTileState.ts:30]: nit: used the -> use

File: src/audiovideocontroller/DefaultAudioVideoController.ts [hunk 2]
```diff
@@ -952,6 +952,7 @@ export default class DefaultAudioVideoController
     if (added.length !== 0 || removed.length !== 0) {
       return false;
     }
+    context.videoStreamIndex.subscribeFrameSent();
```
Comment (devalevenkatesh) [inline src/audiovideocontroller/DefaultAudioVideoController.ts:955]: [For understanding]: So the above `context.signalingClient.remoteVideoUpdate` means sending of a subscribe frame?

Comment (hensmi-amazon) [reply src/audiovideocontroller/DefaultAudioVideoController.ts:955]: Its not actually sending a subscribe, I will update the comment in docstring and above usage. Alternatively I could create an alias function `cacheCurrentIndex` but maybe will save for a new major rev :)

Comment (devalevenkatesh) [reply src/audiovideocontroller/DefaultAudioVideoController.ts:958]: Cool, thanks for explaining and elaborating the docstring.

**Review by devalevenkatesh — APPROVED** [review]

**Pull request merged**